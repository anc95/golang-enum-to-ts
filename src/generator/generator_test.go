package generator

import (
	"os"
	"path"
	"testing"

	"github.com/anc95/golang-enum-to-ts/src/ast"
	"github.com/anc95/golang-enum-to-ts/src/token"
	"github.com/gkampitakis/go-snaps/snaps"
)

func TestNormal(t *testing.T) {
	wd, _ := os.Getwd()

	a := path.Join(wd, "../test-cases/normal.go")
	parser := token.NewParser(a)
	tokens := parser.Parse()
	ast := ast.NewAstGenerator(tokens)

	result := GenerateTS(ast.Gen())

	snaps.MatchSnapshot(t, result)
}
