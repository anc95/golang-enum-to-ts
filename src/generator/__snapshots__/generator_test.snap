
[TestNormal - 1]
namespace some {
  export enum Sex {
    Female = 'female',
    Male = 'male',
  }
  export enum Status {
    Todo = 0,
    Done = 1,
    Pending = 2,
    InProgress = 3,
  }
}

---
